torch/_ops.py

```diff
@@ -234,6 +234,15 @@ def resolve_key(op: OperatorBase, k: DispatchKey):  # type: ignore[valid-type]
 _global_higher_order_ops = {}
 _higher_order_ops = {}
 
+_HIGHER_ORDER_OP_DEFAULT_FALLTHROUGH_DISPATCH_KEYS = [
+    DispatchKey.PythonDispatcher,  # type: ignore[attr-defined]
+    DispatchKey.PythonTLSSnapshot,  # type: ignore[attr-defined]
+    DispatchKey.ADInplaceOrView,
+    DispatchKey.BackendSelect,
+    DispatchKey.AutocastCPU,  # type: ignore[attr-defined]
+    DispatchKey.AutocastCUDA,  # type: ignore[attr-defined]
```

Comment:
Is one still able to py_impl for one of these if they wanted to? (could we add a test for that?)

Reply:
yes, I make HigherOrderOperator override the py_impl method of OperatorBase and will add it to the non_fallthrough_keys if it's in the fallthrough keys list. 

I can add a test.